src/perf/sample/writer.cpp

```diff
@@ -208,9 +208,26 @@ void Writer::end()
     if (cpuid_ == -1)
     {
         // time::now() can sometimes can be in the past :-(
-        last_time_point_ = std::max(last_time_point_, lo2s::time::now());
+        auto now = lo2s::time::now();
+        last_time_point_ = std::max(last_time_point_, now);
+
+        // If we have never written any samples on this location, we also never
+        // got to write the thread_begin event.  Make sure we do that now.
+        if (first_event_)
+        {
+            first_time_point_ = std::min(first_time_point_, now);
```

Comment:
```suggestion
```

This check is pointless. `first_time_point_` will always be smaller (or equal -- veeeeeeery unlikely) than `now`. lo2s::time::now() is a steady clock. We discussed that already in the other PR.

Reply:
*mea culpa.*

Reply:
I'm confused 😕 It's still in?